### PR TITLE
Add generic onnx scaler and ArtCNN scaler

### LINKED ISSUE #89: Implement ArtCNN classes

### PR DESCRIPTION
**Pull request by Vodes**
Fixes #89.

I'll do ArtCNN tomorrow or something.
Let me know if this is fine so far.

Also please put in formatter configs of some sort. My formatter keeps trying to fix every file in this repo.

### CODE REVIEW

File: vsscale/onnx.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            network_path=str(SPath(self.model).resolve()),
+            backend=backend,
+            overlap=(overlap_w, overlap_h),
+            tilesize=(tile_w, tile_h),
+        )
+        scaled = self._finish_scale(scaled, wclip, width, height, shift)
+        return depth(scaled, og_depth)
+
+
+def autoselect_backend() -> Any:
```
Comment (Setsugennoao) [inline vsscale/onnx.py:97]: Should maybe move this with utils and use it in other vsmlrt-based scalers?

File: vsscale/onnx.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        scaled = inference(
+            wclip,
+            network_path=str(SPath(self.model).resolve()),
+            backend=backend,
+            overlap=(overlap_w, overlap_h),
+            tilesize=(tile_w, tile_h),
+        )
+        scaled = self._finish_scale(scaled, wclip, width, height, shift)
+        return depth(scaled, og_depth)
```
Comment (Setsugennoao) [inline vsscale/onnx.py:94]: finish_scale already handles this if you pass clip instead of wclip.

**Review by Setsugennoao — CHANGES_REQUESTED** [review]

File: vsscale/onnx.py (hunk 2 above)
Comment (Vodes) [reply vsscale/onnx.py:94]: That's a remnant from me using functionutil before and then noticing I dont really need it; Will remove

File: vsscale/onnx.py (hunk 1 above)
Comment (Vodes) [reply vsscale/onnx.py]: For now I'll let you handle this probably. I don't know how you'd want that to integrate with w2x considering thats the one other vsmlrt based scaler in here and some of those params surely aren't too good for every generic model.

**Pull request merged**